fabtests/common/shared.c

```diff
@@ -578,18 +597,14 @@ int ft_alloc_msgs(void)
 		if (ret)
 			return ret;
 	}
+
 	ret = ft_hmem_memset(opts.iface, opts.device, (void *) buf, 0, buf_size);
 	if (ret)
 		return ret;
-	rx_buf = buf;
-
-	if (opts.options & FT_OPT_ALLOC_MULT_MR)
-		tx_buf = (char *) buf + MAX(rx_size, FT_MAX_CTRL_MSG);
-	else
-		tx_buf = (char *) buf +
-			 MAX(rx_size, FT_MAX_CTRL_MSG) * opts.window_size +
-			 rma_resv_bytes;
 
+	rx_buf = (char *)ft_get_aligned_addr(buf, alignment);
+	assert((rx_buf - buf) >= 0 && (rx_buf - buf) < alignment);
```

Comment:
I think it's better to assert `alignment > 0` in `ft_get_aligned_addr` than to do it here

Reply:
Agree that asserts should be moved into the macro

Reply:
The assertion here is to make sure the returned buffer is equal to or larger than input. Because the function name `ft_get_aligned_addr` does not tell caller which way it goes

Reply:
I'd go further with this and match std::align (somewhat).  E.g.
```
ft_get_aligned_addr(void *buf, size_t buf_size, size_t alignment, size_t *new_size)
```
This helps clarify that we're aligning into the buffer and the returned buffer has fewer bytes available.

Reply:
I do not follow. What do we do with `new_size` returned from this function? add an asserion that `new_size <= size`?

Reply:
There are a couple options.

The new_size is the size of what we can access starting at rx_buf.  Technically, our memory registration (if needed) starts at rx_buf and extends to new_size.  Or, if we want to be more precise, we expect rx_buf to be usable to size half_buf_size (I think?).  Then tx_buf would start at rx_buf + half_buf_size, and the size of the tx_buf would be new_size - half_buf_size.  The memory registration for rx_buf and tx_buf could be separated.

What I would assert is that new_size >= the size that we need.  I'm not sure what values give us that.  (new_size >= tx_size + rx_size) ?  (new_size >= MAX(tx_size, rx_size)) ?

Or maybe (rx_size < new_size - half_buf_size) && (tx_size <= new_size - half_buf_size) ?

Note that ft_get_aligned_addr() could fail (return null) if the buffer size isn't large enough to meet the alignment requirement.  For example, buf_size = 128, but alignment = 4k, could fail.